pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -919,6 +919,11 @@ func (p *PriorityQueue) Update(logger klog.Logger, oldPod, newPod *v1.Pod) error
 	p.lock.Lock()
 	defer p.lock.Unlock()
 
+	if _, ok := p.inFlightPods[newPod.UID]; ok {
```

Comment:
If we just skip the update here, the old Pod object, which is being processed, will come back to the scheduling queue and stay there. That Pod won't be up-to-date until that Pod gets updated again -- meaning the scheduler keeps handling the Pod with the old spec/state.

So, can we somehow avoid that? Probably we should record the latest Pod spec somewhere so that we can update the Pod spec when the Pod comes back from the scheduling cycle.

My idea is to use `p.inFlightPods` to store the latest Pod spec for in-flight Pods.
- If the updated Pod is in-flight, replace `p.inFlightPods[newPod.UID].Value` with the updated Pod. 
  - ```
    // https://github.com/kubernetes/kubernetes/blob/9e44875c45d08fbf1a99ba24604d2e8a60f3581f/pkg/scheduler/internal/queue/scheduling_queue.go#L923-L924

    if _, ok := p.inFlightPods[newPod.UID]; ok {
       p.inFlightPods[newPod.UID].Value = newPod
    ``` 
- When the Pod comes back to the scheduling queue, enqueue the Pod stored in `p.inFlightPods[pod.UID].Value`.
  - ```
    // https://github.com/kubernetes/kubernetes/blob/9e44875c45d08fbf1a99ba24604d2e8a60f3581f/pkg/scheduler/internal/queue/scheduling_queue.go#L751
    
    pInfo.Pod = p.inFlightPods[pod.UID].Value.(*v1.Pod)
    queue := p.requeuePodViaQueueingHint(logger, pInfo, schedulingHint, ScheduleAttemptFailure)
    ```

Reply:
>If we just skip the update here, the old Pod object, which is being processed, will come back to the scheduling queue and stay there. That Pod won't be up-to-date until that Pod gets updated again -- meaning the scheduler keeps handling the Pod with the old spec/state.

If a pod is unschedule, when it put back, We will get the new pod from informer, so I think the pod will be updated to the latest status

https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/schedule_one.go#L1005-L1026

Reply:
Oh, that's a good point. Forget what I was saying.